mpmp/prediction/cross_validation.py

```diff
@@ -89,6 +91,17 @@ def run_cv_stratified(data_model,
         y_train_df = data_model.y_df.reindex(X_train_raw_df.index)
         y_test_df = data_model.y_df.reindex(X_test_raw_df.index)
 
+        # shuffle labels for train/test sets separately
+        # this ensures that overall label balance isn't affected
+        # (see https://github.com/greenelab/mpmp/issues/44)
+        if shuffle_labels:
+            # we set a temp seed here to make sure this shuffling order
+            # is the same for each gene between data types, otherwise
+            # it might be slightly different depending on the global state
+            with temp_seed(data_model.seed):
+                y_train_df.status = np.random.permutation(y_train_df.status.values)
+                y_test_df.status = np.random.permutation(y_test_df.status.values)
```

Comment:
Ooh this is really cool. I'll have to remember this logic for shuffling things in the future

Reply:
Yeah, it's useful!